packages/python/plotly/plotly/tests/test_core/test_px/test_px_input.py

```diff
@@ -0,0 +1,267 @@
+import plotly.express as px
+import numpy as np
+import pandas as pd
+import pytest
+import plotly.graph_objects as go
+import plotly
+from plotly.express._core import build_dataframe
+from pandas.util.testing import assert_frame_equal
+
+attrables = (
+    ["x", "y", "z", "a", "b", "c", "r", "theta", "size", "dimensions"]
+    + ["custom_data", "hover_name", "hover_data", "text"]
+    + ["error_x", "error_x_minus"]
+    + ["error_y", "error_y_minus", "error_z", "error_z_minus"]
+    + ["lat", "lon", "locations", "animation_group"]
+)
+array_attrables = ["dimensions", "custom_data", "hover_data"]
+group_attrables = ["animation_frame", "facet_row", "facet_col", "line_group"]
+
+all_attrables = attrables + group_attrables + ["color"]
+
+
+def test_numpy():
+    fig = px.scatter(x=[1, 2, 3], y=[2, 3, 4], color=[1, 3, 9])
+
+
+def test_numpy_labels():
+    fig = px.scatter(
+        x=[1, 2, 3], y=[2, 3, 4], labels={"x": "time"}
+    )  # other labels will be kw arguments
+    assert fig.data[0]["hovertemplate"] == "time=%{x}<br>y=%{y}"
+
+
+def test_with_index():
+    tips = px.data.tips()
+    fig = px.scatter(tips, x=tips.index, y="total_bill")
+    assert fig.data[0]["hovertemplate"] == "index=%{x}<br>total_bill=%{y}"
+    fig = px.scatter(tips, x=tips.index, y=tips.total_bill)
+    assert fig.data[0]["hovertemplate"] == "index=%{x}<br>total_bill=%{y}"
+    # If we tinker with the column then the name is the one of the kw argument
+    fig = px.scatter(tips, x=tips.index, y=10 * tips.total_bill)
+    assert fig.data[0]["hovertemplate"] == "index=%{x}<br>y=%{y}"
+    fig = px.scatter(tips, x=tips.index, y=tips.total_bill, labels={"index": "number"})
+    assert fig.data[0]["hovertemplate"] == "number=%{x}<br>total_bill=%{y}"
+    # We do not allow "x=index"
+    with pytest.raises(ValueError) as err_msg:
+        fig = px.scatter(tips, x="index", y="total_bill")
+        assert (
+            "ValueError: Value of 'x' is not the name of a column in 'data_frame'"
+            in str(err_msg.value)
+        )
+
+
+def test_pandas_series():
+    tips = px.data.tips()
+    before_tip = tips.total_bill - tips.tip
+    fig = px.bar(tips, x="day", y=before_tip, labels={"y": "bill"})
+    assert fig.data[0].hovertemplate == "day=%{x}<br>bill=%{y}"
+
+
+def test_several_dataframes():
+    df = pd.DataFrame(dict(x=[0, 1], y=[1, 10], z=[0.1, 0.8]))
+    df2 = pd.DataFrame(dict(time=[23, 26], money=[100, 200]))
+    fig = px.scatter(df, x="z", y=df2.money, size="y")
+    assert fig.data[0].hovertemplate == "z=%{x}<br>y_money=%{y}<br>y=%{marker.size}"
```

Comment:
this result surprises me a bit...

Reply:
why? It's a bit weird but it corresponds to the logics
- use the field name if possible
- if not use field + _ + name
Hopefully users will use sensible column names most of the time...

Reply:
this has been chnged, it would raise an error